nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-core/src/test/java/org/nuxeo/ecm/directory/core/TestCoreDirectory.java

```diff
@@ -142,10 +142,20 @@ public void testUpdateEntry() throws Exception {
 
     @Test
     public void testAuthenticate() throws Exception {
-        Assert.assertTrue(dirSession.authenticate(CoreDirectoryInit.DOC_ID_USER1,
-                CoreDirectoryInit.DOC_PWD_USER1));
+        Assert.assertTrue(dirSession.authenticate(CoreDirectoryInit.DOC_ID_USER1, CoreDirectoryInit.DOC_PWD_USER1));
         Assert.assertFalse(dirSession.authenticate(CoreDirectoryInit.DOC_ID_USER1, "bad-pwd"));
         Assert.assertFalse(dirSession.authenticate("bad-id", "haha"));
+        Assert.assertTrue(
+                dirSession.authenticate(CoreDirectoryInit.DOC_ID_USERSHA1, CoreDirectoryInit.DOC_PWD_USERSHA1)); // password
+                                                                                                                 // alright
+        Assert.assertFalse(
+                dirSession.authenticate(CoreDirectoryInit.DOC_ID_USERSHA1, CoreDirectoryInit.DOC_PWD_BADPWDSHA1)); // wrong
+                                                                                                                   // password
+        Assert.assertFalse(
+                dirSession.authenticate(CoreDirectoryInit.DOC_ID_USERSHA1, CoreDirectoryInit.DOC_PWD_NULLSHA1)); // null
+                                                                                                                 // password
+                                                                                                                 // (avoid
+                                                                                                                 // NPE)
```

Comment:
please put comments just above the assert instead of the end of the line: it'll be more readable